### PR TITLE
Access to api setAuthenticatedUserContext.

### PR DESCRIPTION
**Pull request by jpiyali**

### CODE REVIEW

File: src/ReactAI.ts [hunk 1]
```diff
@@ -1,32 +1,27 @@
...
+ * @class ReactAI  
  */
 class ReactAI implements ITelemetryPlugin {
   public extensionId: string = "ApplicationInsightsReactUsage";
   public ApplicationInsightsAnalyticsIdentifier: string = "ApplicationInsightsAnalytics";
   public processTelemetry: (env: ITelemetryItem) => void;
   public identifier = this.extensionId;
   public priority: number = 190;
-  public appInsights!: IApplicationInsights;
+  public appInsights!: AppInsightsPlugin;
```
Comment (hiraldesai) [inline src/ReactAI.ts:23]: You want this to be public for the consumers to be able to access?

Comment (jpiyali) [reply src/ReactAI.ts:23]: There is nothing private about the internal IAppInsights. the public layer is a combination of multiple of such interfaces.

**Comment by pviotti** [comment]
Thanks for this other PR! 
The TS compiler complains about this:
```
src/ReactAI.ts:132:11 - error TS7017: Element implicitly has an 'any' type because type 'ICustomProperties' has no index signature.
```
A simple and quick fix would be having
```typescript
(envelope.data as any)[key] = props[key];
```
on line 132. But I suspect there are better ways than giving up on typing...

File: src/ReactAI.ts [hunk 2]
```diff
@@ -1,32 +1,27 @@
...
+  private propertiesPlugin!: PropertiesPlugin;
```
Comment (hiraldesai) [inline src/ReactAI.ts:24]: I apologize my previous comment was about `propertiesPlugin` being private. Shouldn't it be public? If it's private, what is using it, and why does it need to be there in the first place?

Comment (jpiyali) [reply src/ReactAI.ts:24]: You need it to access the setAuthenticatedUserContext() api

Comment (jpiyali) [reply src/ReactAI.ts:24]: I dont have the code change here that you had for accessing that api which was the original issue.

**Comment by hiraldesai** [comment]
Still don’t know what a private propertiesplugin is for. But we will just merge and test it out.

**Pull request merged**